sklearn/datasets/openml.py

```diff
@@ -424,17 +424,40 @@ def _get_data_qualities(data_id, data_home):
         return None
 
 
-def _get_data_shape(data_qualities):
-    # Using the data_info dictionary from _get_data_info_by_name to extract
-    # the number of samples / features
+def _get_data_shape(data_qualities, features_list):
```

Comment:
Since we are now using `len(features_list)` for all length, this function can be refactored to be just `_get_data_instances`, which returns the actual number of instances or `-1` when it does not know.